lib/moonshot/stack.rb

```diff
@@ -154,12 +155,14 @@ def default_values
     end
 
     def template
-      @template ||= StackTemplate.new(template_file, log: @log)
+      @template ||= load_template_file
     end
 
     # @return [String] the path to the template file.
     def template_file
-      File.join(Dir.pwd, 'cloud_formation', "#{@app_name}.json")
+      json = "#{raw_template_file_name}.json"
+      yaml = "#{raw_template_file_name}.y*ml"
+      @template_file ||= Dir[json].first || Dir[yaml].first || raise
```

Comment:
If so, I think you could use `load_template_file.filename` here, if you make an attr_reader on `@filename`.